hw/mcu/dialog/da1469x/src/da1469x_lpclk.c

```diff
@@ -57,11 +63,18 @@ da1469x_lpclk_register_cmac_cb(da1469x_lpclk_cb *cb)
 void
 da1469x_lpclk_init(void)
 {
+#if MYNEWT_VAL(MCU_CLOCK_XTAL32K_SWITCH)
     static struct hal_timer lpclk_settle_tmr;
-
     da1469x_clock_lp_xtal32k_enable();
-
     os_cputime_timer_init(&lpclk_settle_tmr, da1469x_lpclk_settle_tmr_cb, NULL);
     os_cputime_timer_relative(&lpclk_settle_tmr,
                               MYNEWT_VAL(MCU_CLOCK_XTAL32K_SETTLE_TIME_MS) * 1000);
+#endif
+#if MYNEWT_VAL(MCU_CLOCK_RCX_SWITCH)
+    static struct hal_timer lpclk_settle_tmr;
+    da1469x_clock_lp_rcx_enable();
+    os_cputime_timer_init(&lpclk_settle_tmr, da1469x_lpclk_settle_tmr_cb, NULL);
+    os_cputime_timer_relative(&lpclk_settle_tmr,
+                              MYNEWT_VAL(MCU_CLOCK_RCX_SETTLE_TIME_MS) * 1000);
+#endif
```

Comment:
I presume that either MCU_CLOCK_RCX_SWITCH or MCU_CLOCK_XTAL32K should be defined. If so, I think there should be some error if both are set to 0. Not sure if there is somewhere but if not it should be added.